src/components/MapView/Legends/index.tsx

```diff
@@ -76,35 +89,113 @@ function Legends({ classes, layers }: LegendsProps) {
 }
 
 // Children here is legendText
-function LegendItem({ classes, title, legend, children }: LegendItemProps) {
+function LegendItem({
+  classes,
+  id,
+  title,
+  legend,
+  type,
+  opacity: initialOpacity,
+  children,
+}: LegendItemProps) {
+  const map = useSelector(mapSelector);
+  const [opacity, setOpacityValue] = useState<number | number[]>(
+    initialOpacity || 0,
+  );
+
+  const [toggle, setToggle] = useState(true);
+  const toggleChecked = (prev: Boolean) => {
+    setToggle(!prev);
+  };
+
+  const handleChangeOpacity = (event: object, newValue: number | number[]) => {
+    setOpacityValue(newValue);
+  };
+
+  const layerTypes = ['wms', 'nso', 'point_data', 'impact'];
```

Comment:
You should type this so that as layer types change, this updates too. I don't remember the codebase too well but I think there might be an enum somewhere with these values. Try looking for it, otherwise just a typing here to remind us to update it if layer types change would be nice.